lib/ssh_scan/api.rb

```diff
@@ -162,8 +169,25 @@ class API < Sinatra::Base
         if worker_id.empty? || uuid.empty?
           return {"accepted" => "false"}.to_json
         end
+        results = JSON.parse(request.body.first).first
+        settings.current_avg = settings.current_avg +
+        (results['scan_duration_seconds'] - settings.current_avg) / settings.count_req
```

Comment:
Even if you want to do it this way, shouldn't be the `new_avg = (old_avg*n + time_spent_on_curr_task) / (n+1)`? ,i.e.,
`settings.current_avg = (settings.current_avg*(settings.count_req-1) + results['scan_duration_seconds'] - settings.current_avg) / settings.count_req`